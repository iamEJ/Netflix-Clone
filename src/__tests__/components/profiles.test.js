import React from "react";
import { render, getByTestId } from "@testing-library/react";
import Profiles from "../../components/profiles";

describe("<Profiles /> ", () => {
  it("renders <Profiles /> with populated data", () => {
    const { container, getByText, getByTestId } = render(
      <Profiles>
        <Profiles.Title>Who's watching?</Profiles.Title>
        <Profiles.List>
          <Profiles.Item onClick={() => {}}>
            <Profiles.Picture
              src="/images/profile.png"
              data-testid="profile-image"
            />
            <Profiles.Name>Jenny</Profiles.Name>
          </Profiles.Item>
        </Profiles.List>
      </Profiles>
    );
    expect(getByText("Who's watching?")).toBeTruthy();
    expect(getByTestId("profile-image")).toBeTruthy();
    expect(getByText("Jenny")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
