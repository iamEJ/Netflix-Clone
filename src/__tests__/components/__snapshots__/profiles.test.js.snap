// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profiles />  renders <Profiles /> with populated data 1`] = `
<div
  class="profiles__Container-sc-1b9qd60-0 bkKYqI"
>
  <h1
    class="profiles__Title-sc-1b9qd60-1 igUsWg"
  >
    Who's watching?
  </h1>
  <ul
    class="profiles__List-sc-1b9qd60-2 iqPkYr"
  >
    <li
      class="profiles__Item-sc-1b9qd60-5 gbFRBD"
    >
      <img
        class="profiles__Picture-sc-1b9qd60-3 jMcbgH"
        data-testid="profile-image"
        src="https://mir-s3-cdn-cf.behance.net/project_modules/disp/64623a33850498.56ba69ac2a6f7.png"
      />
      <p
        class="profiles__Name-sc-1b9qd60-4 gEuEtI"
      >
        Jenny
      </p>
    </li>
  </ul>
</div>
`;
